scripts/utilities.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+
+import argparse
+import logging
+import os
+import sys
+
+from typing import List, Optional
+
+
+logger = logging.getLogger(__name__)
+
+
+DATABASES = ["panther", "ncbifam", "cathgene3d"]
+
+
+class ValidateDatabases(argparse.Action):
+    """Check the user has provided valid databases."""
+    def __call__(self, parser, args, values, option_string=None):
+        invalid = False
+        for value in values:
+            if value.lower() not in DATABASES:
+                invalid = True
+                raise ValueError(f'Invalid database "{value}" provided. Accepted databases: {DATABASES}')
+        if invalid:
+            sys.exit(1)
+        setattr(args, self.dest, [value.lower() for value in values])
+
+
+def clean_url(url):
+    return url.rstrip("/")
+
+
+def build_parser(argv: Optional[List] = None):
+    parser = argparse.ArgumentParser(
+        prog="Import AI generated data",
+        description="Automatically integrated AI-generated signatures into the IPPRO database",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+
+    parser.add_argument(
+        "-b",
+        "--batch_size",
+        type=int,
+        default=50000000,
+        help="Number of signatures to parse at a time.",
+    )
+
+    parser.add_argument(
+        "-c",
+        "--cache",
+        action="store_true",
+        default=False,
+        help="Write out signatures with non-unique name/short name to a TSV file",
+    )
+
+    parser.add_argument(
+        "--databases",
+        action=ValidateDatabases,
```

Comment:
I guess that you're trying to handle database names in a case insensitive way, but it's much simpler with `type=str.lower`:

```suggestion
        type=str.lower,
```